Project1/Mobility.cpp

```diff
@@ -28,6 +30,18 @@ int& Mobility::getMove() {
 	return _movement;
 }
 
-bool Mobility::canPass(MobilityType mobility_type, const Unit* unit) const {
-	return _mobility[mobility_type] && (unit == nullptr ||_owner._party->hasUnit(*unit));
-}
\ No newline at end of file
+bool Mobility::canPass(const Unit* unit) const {
+	//TODO(Torrey): add skill Logic stuff
+	return unit == nullptr || _owner._party->hasUnit(*unit);
+}
+
+std::optional<int> Mobility::getCost(const CellEdge & edge) const {
+	std::optional<int> cost;
+	for (MobilityType type : MobilityType::list) {
+		std::optional<int> edge_cost = edge.getCost(type);
+		if (_mobility[type] && canPass(edge._cell.getTile()._unit) && edge_cost.has_value() && (cost > edge_cost || !cost.has_value())) { // if we can step on the tile
```

Comment:
```
if (_mobility[type] && canPass(edge._cell.getTile()._unit)) {
    std::optional<int> edge_cost = edge.getCost(type);
    if (!cost || cost > edge_cost) {
        cost = edge_cost.value();
    }
}
````

You can swap the second if's condition with `edge_cost && (cost > edge_cost || !cost)` if you want to avoid assigning None to cost when it's already None, but that 5-condition if needs breaking up especially because you don't need edge_cost if it fails early.

Reply:
canPass(edge._cell.getTile()._unit) is going to be removed anyway since Tile will be losing their unit, but I fixed this anway